template/shank/base/program/src/assertions.rs

```diff
@@ -4,6 +4,26 @@ use solana_program::{
     pubkey::Pubkey,
 };
 
+/// Assert that the given account is owned by the given program or one of the given owners.
+/// Useful for dealing with program interfaces.
+pub fn assert_program_owner_either(
+    account_name: &str,
+    account: &AccountInfo,
+    owners: &[Pubkey],
+) -> ProgramResult {
+    if !owners.iter().any(|owner| account.owner == owner) {
+        msg!(
+            "Account \"{}\" [{}] must be owned by either {:?}",
```

Comment:
While this is visually the best way to format the message, logging a pubkey this way is super expensive (easily > 5k CU). Perhaps it would be better to have an example on how to log the pubkey in a CU efficient way using the `Pubkey::log` method, which takes "only" ~100 CU.

Something along the lines:
```rust
msg!("Account \"{}\":", account_name);
account.key.log();
msg!("must be owned by either:");
owners.iter().for_each(|owner| owner.log());
```

Reply:
Thanks for this Febo. Just wanted to flag something I've just noticed: The other helpers are already logging pubkeys in an efficient way. Perhaps we could merge this as-is, and try and improve the logging efficiently in a subsequent PR? Or just leave it inefficient until we replace these helpers with the `!require` macro as we previously discussed?

Reply:
Yeah, happy with that. Let's merge this and do another PR that changes all pubkey logs.